pydantic/main.py

```diff
@@ -169,6 +171,60 @@ def validate_custom_root_type(fields: Dict[str, Field]) -> None:
 TYPE_BLACKLIST = FunctionType, property, type, classmethod, staticmethod
 
 
+class ValueExclude:
+    __slots__ = ('__exclude__', '__type__')
+
+    def __init__(self, value: Any, exclude: Union['SetIntStr', 'DictIntStrAny']) -> None:
+        if TYPE_CHECKING:
+            self.__exclude__: Union['SetIntStr', 'DictIntStrAny']
+            self.__type__: Type[Union[set, dict]]  # type: ignore
+
+        self.__exclude__ = exclude
+
+        # For further type check speed-up
+        if isinstance(exclude, dict):
+            self.__type__ = dict
```

Comment:
can we use an enum rather than `dict` or `set`

Reply:
Wouldn't it be slower?
See: https://github.com/Rapptz/discord.py/commit/991140eebe4a618354ada9ec592e9a405dab604d#diff-e42d1e46f30339be8fb6f28250923015

Reply:
yes, I'd have

```py
class DictSet(IntEnum):
    dict = 1
    set = 2
```

Reply:
And use like this? 
```py
if isinstance(exclude, dict):
    self.__type__ = DictOrSet.dict

...

if self.__type__ is DictOrSet.dict:
    ...

```
Still can't get the point here :(

Reply:
yes.

Reply:
should be faster since we're just comparing ints not types.

Reply:
Nooo, enum is much slower since getattr is used, and we comparing not ints, but enum instances

Reply:
```py
from enum import IntEnum
from timeit import timeit


class DictOrSet(IntEnum):
    dict = 1
    set = 2

_type = dict
_enum_type = DictOrSet.dict


t0 = timeit('_type is dict', globals={'_type': _type}, number=50000000)
t1 = timeit('_enum_type is DictOrSet.dict', globals={'_enum_type': _enum_type, 'DictOrSet': DictOrSet}, number=50000000)

print(t0)  #  1.602347916999861
print(t1)  #  4.7232167160000245
```

Reply:
ok.

Reply:
yes, just did the exact same thing.